kapre/augmentation.py

```diff
@@ -101,3 +106,183 @@ def get_config(self):
+        mask_value (`float`): Value of the applied masks. By default is 0.
+        data_format (`str`): specifies the data format of batch input/output
+        **kwargs: Keyword args for the parent keras layer (e.g., `name`)
+
+    Example:
+        ::
+
+            input_shape = (2048, 1)  # mono signal
+
+            # We compute the Mel Spectrogram of the input signal
+            melgram = kapre.composed.get_melspectrogram_layer(input_shape=input_shape,
+                                                  n_fft=1024,
+                                                  return_decibel=True,
+                                                  n_mels=256,
+                                                  input_data_format='channels_last',
+                                                  output_data_format='channels_last')
+
+
+            # Now we define the SpecAugment layer. It will apply 5 masks in the frequency axis,
+            # 3 masks in the time axis. The frequency mask param is 5 and the time mask param
+            # is 10.
+            spec_augment = SpecAugment(freq_mask_param=5,
+                                       time_mask_param=10,
+                                       n_freq_masks=5,
+                                       n_time_masks=3)
+
+            model = Sequential()
+            model.add(melgram)
+
+            # Add the spec_augment layer for augmentation
+            model.add(spec_augment)
+        ::
+    """
+
+    def __init__(self,
+                 freq_mask_param,
+                 time_mask_param,
+                 n_freq_masks=1,
+                 n_time_masks=1,
+                 mask_value=0.,
+                 data_format='default',
+                 **kwargs
+                 ):
+
+        backend.validate_data_format_str(data_format)
+
+        super(SpecAugment, self).__init__(**kwargs)
+
+        self.freq_mask_param = freq_mask_param
+        self.time_mask_param = time_mask_param
+        self.n_freq_masks = n_freq_masks
+        self.n_time_masks = n_time_masks
+        self.mask_value = mask_value
+
+        self.data_format = K.image_data_format() if data_format == _CH_DEFAULT_STR else data_format
+
+    @staticmethod
+    def _generate_axis_mask(inputs):
+        """
+        Generate a mask for the axis provided
+        Args:
+            inputs (`tuple`): A 3-tuple with the following structure:
+                inputs[0]: A spectrogram
+                inputs[1]: The axis where the mask will be applied
+                inputs[2]: The mask param as defined in the original paper
+        Returns:
+            A mask represented as a boolean tensor
+        """
+        x, axis, mask_param = inputs
+
+        axis_limit = tf.shape(x)[axis]
+        axis_indices = tf.range(axis_limit)
+
+        if axis == 0:
+            axis_indices = tf.reshape(axis_indices, (-1, 1, 1))
+        elif axis == 1:
+            axis_indices = tf.reshape(axis_indices, (1, -1, 1))
+        else:
```

Comment:
would `else` be okay? or shall we add `elif axis ==2:` followed by `else: raise NotImplementedError`? i usually prefer the latter unless it's taken care of somewhere else very clearly.